pkg/auth/handler/wire.go

```diff
@@ -0,0 +1,37 @@
+//+build wireinject
+
+package handler
+
+import (
+	"net/http"
+
+	"github.com/google/wire"
+
+	"github.com/skygeario/skygear-server/pkg/core/config"
+)
+
+func ProvideTenantConfigPtr(r *http.Request) *config.TenantConfiguration {
+	return config.GetTenantConfig(r.Context())
+}
+
+func ProvideTenantConfig(r *http.Request) config.TenantConfiguration {
```

Comment:
If we're refactoring/changing the DI mechanism, we may as well as unify to use either pointer/value, but not both. Is that possible?

Reply:
Yes

Reply:
Removed the provider of struct and added notes.